src/lib.rs

```diff
@@ -594,53 +604,109 @@ impl TryFrom<&[u8]> for ScriptPubkey {
     }
 }
 
+impl Clone for ScriptPubkey {
+    fn clone(&self) -> Self {
+        ScriptPubkey {
+            inner: unsafe { btck_script_pubkey_copy(self.inner) },
+        }
+    }
+}
+
 impl Drop for ScriptPubkey {
     fn drop(&mut self) {
-        unsafe { kernel_script_pubkey_destroy(self.inner) }
+        unsafe { btck_script_pubkey_destroy(self.inner) }
     }
 }
 
-/// A single transaction output.
+/// A single transaction output containing a value and spending conditions.
 ///
-/// It can be initialized with a script pubkey and amount, and the user may
-/// retrieve a copy of a script pubkey and its amount.
-#[derive(Debug, Clone)]
+/// Transaction outputs can be created from a script pubkey and amount, or retrieved
+/// from existing transactions. They represent spendable coins in the UTXO set.
+#[derive(Debug)]
 pub struct TxOut {
-    inner: *mut kernel_TransactionOutput,
+    inner: *mut btck_TransactionOutput,
 }
 
 unsafe impl Send for TxOut {}
 unsafe impl Sync for TxOut {}
 
 impl TxOut {
+    /// Creates a new transaction output with the specified script and amount.
+    ///
+    /// # Arguments
+    /// * `script_pubkey` - The script defining how this output can be spent
+    /// * `amount` - The amount in satoshis
     pub fn new(script_pubkey: &ScriptPubkey, amount: i64) -> TxOut {
         TxOut {
-            inner: unsafe { kernel_transaction_output_create(script_pubkey.inner, amount) },
+            inner: unsafe { btck_transaction_output_create(script_pubkey.inner, amount) },
         }
     }
 
-    /// Get the amount associated with this transaction output
-    pub fn get_value(&self) -> i64 {
-        unsafe { kernel_transaction_output_get_amount(self.inner) }
+    pub(crate) fn from_raw(ptr: *mut btck_TransactionOutput) -> TxOut {
+        TxOut { inner: ptr }
+    }
+
+    /// Returns the amount of this output in satoshis.
+    pub fn value(&self) -> i64 {
+        unsafe { btck_transaction_output_get_amount(self.inner) }
     }
 
-    /// Get the script pubkey of this output
-    pub fn get_script_pubkey(&self) -> ScriptPubkey {
+    /// Returns the script pubkey that defines how this output can be spent.
+    ///
+    /// The returned script pubkey is independent of this output's lifetime.
```

Comment:
I don't think this is true, the returned script pubkey here should still be dependent on the output's lifetime. Or am I missing something? I think this similarly needs to either return a ref type, manage the lifetime explicitly, or add phantom data.